src/app/lists/editModal.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+
+import Modal from "@/components/Modal/Modal";
+import React, { useState } from "react";
+import { styled } from "styled-components";
+import { SnackBarDiv, headers, tooltips } from "@/app/lists/dataview";
+import TextInput from "@/components/DataInput/FreeFormTextInput";
+import { Datum } from "@/components/Tables/Table";
+import supabase from "@/supabase";
+import Snackbar from "@mui/material/Snackbar/Snackbar";
+import Alert from "@mui/material/Alert/Alert";
+import Button from "@mui/material/Button/Button";
+
+type Props = {
+    onClose: () => void;
+    /** null => add, undefined => modal closed, Datum => edit */
+    data: Datum | null | undefined;
+};
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+`;
+
+const DataWithTooltipDiv = styled.div`
+    display: flex;
+    gap: 1rem;
+    flex-direction: column;
+
+    @media (min-width: 600px) {
+        flex-direction: row;
+    }
+`;
+
+const DisplayContents = styled.div`
+    display: contents;
+`;
+
+const EditModal: React.FC<Props> = ({ data, onClose }) => {
+    const generateDefaultData = (): { [key: string]: string | null } => {
+        if (data !== null && data !== undefined) {
+            return { ...data.data, ...data.tooltips };
+        }
+        return {};
+    };
+
+    const [toSubmit, setToSubmit] = useState<{ [key: string]: string | null }>(
+        generateDefaultData()
+    );
+
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    const headersAndTooltips: [[string, string], [string, string]][] = [];
+    const headersThatHaveTooltips = headers.filter(([key]) => key !== "item_name");
+
+    for (let index = 0; index < headersThatHaveTooltips.length; index++) {
+        headersAndTooltips.push([headersThatHaveTooltips[index], tooltips[index]]);
+    }
+
+    return (
+        <Modal header="Edit List" headerId="editList" isOpen={data !== undefined} onClose={onClose}>
+            <ModalInner>
+                <h3>Description</h3>
+                <TextInput
+                    defaultValue={toSubmit.item_name ?? ""}
+                    onChange={(event) => {
+                        const text = event.target.value as string;
```

Comment:
this is used enough that it may be worth extracting into a separate function - take specifically a text input on change event and the key to change, and set the state of the object